lib/logstash/outputs/opsgenie.rb

```diff
@@ -174,9 +195,32 @@ def executePost(uri, params)
         http.use_ssl = true
         http.verify_mode = OpenSSL::SSL::VERIFY_NONE
       end
-      request = Net::HTTP::Post.new(url.request_uri, initheader = { "Content-Type" =>"application/json", "Authorization" => "GenieKey #{@apiKey}" })
-      request.body = params.to_json
-      response = http.request(request)
+
+      retryCount = 0
+      begin
+        request = Net::HTTP::Post.new(
+          url.request_uri,
+          initheader = { "Content-Type" =>"application/json", "Authorization" => "GenieKey #{@apiKey}" }
+        )
+        request.body = params.to_json
+        response = http.request(request)
+      rescue *HTTP_EXCEPTIONS => e
+        @logger.warn("Silencing exception in logstash opsgenie outputs: [#{e.class}: #{e.message}]")
+        return
+      rescue *RETRYABLE_HTTP_EXCEPTIONS => e
+        if retryCount < HTTP_EXCEPTION_RETRIES
+          retryCount += 1
```

Comment:
Note that with this setup you do a total of 6 `HTTP_EXCEPTION_RETRIES + 1` _attempts_. I.e. we will also do the request when `retryCount == HTTP_EXCEPTION_RETRIES`.

Not sure if this was intentional or not so thought I'd mention it. Can easily be fixed by moving the increment statement to before the `if` check. In that case we might want to consider renaming it to `tryCount` or something as it would be the count of total number of attempts, not just retries.

Reply:
yes, I know, it is first attempt + 5 retries. Do you think it is too many? total time should be around 31s.